### PR TITLE
Added class-level @getter annotation in AbstractHealthCheckEnricher.Config

### LINKED ISSUE #3402: AbstractHealthCheckEnricher : Use class level `@Getter` annotation in `AbstractHealthCheckEnricher.Config` nested class
### Component

JKube Kit

### Description
AbstractHealthCheckEnricher contains a nested class Config

https://github.com/eclipse-jkube/jkube/blob/dad4cc61149c435dbf91a5700a2c7f7cfe7d9a99/jkube-kit/enricher/specific/src/main/java/org/eclipse/jkube/kit/enricher/specific/AbstractHealthCheckEnricher.java#L42-L51

There are only two fields in this class, both annotated with[`@Getter`](https://projectlombok.org/features/GetterSetter) , if all class members need to have [`@Getter`](https://projectlombok.org/features/GetterSetter) 
annotation, we can move the annotation to class level.

### Expected Behavior
[`@Getter`](https://projectlombok.org/features/GetterSetter) annotation is moved from field level to class level

### Acceptance Criteria
- [x] [`@Getter`](https://projectlombok.org/features/GetterSetter)  is removed from fields and added on top of class declaration
- [x] AbstractHealthCheckEnricherTest is passing after making these changes

### PR DESCRIPTION
**Pull request by vikasrajputin**
## Description
Added class-level @getter annotation in AbstractHealthCheckEnricher.Config

Added 

Fixes #3402 

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [x] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [x] I have added tests that prove my fix is effective or that my feature works
 - [x] New and existing unit tests pass locally with my changes
 - [x] I tested my code in Kubernetes
 - [x] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/eclipse-jkube/ci)

Started new GH workflow run for https://github.com/eclipse-jkube/jkube/pull/3429 (_2024-10-06T10:28:58Z_)

:gear: [JKube E2E Tests (11201088998)](https://github.com/eclipse-jkube/ci/actions/runs/11201088998) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#3429)](https://github.com/eclipse-jkube/ci/actions/runs/11201088998/job/31135597430)

File: gradle-plugin/kubernetes/src/test/java/org/eclipse/jkube/gradle/plugin/task/KubernetesBuildTaskTest.java [hunk 1]
```diff
@@ -56,7 +56,7 @@ class KubernetesBuildTaskTest {
   private boolean isBuildError;
 
   @BeforeEach
-  void setUp() throws IOException {
+  void setUp() {
```
Comment (rohanKanojia) [inline gradle-plugin/kubernetes/src/test/java/org/eclipse/jkube/gradle/plugin/task/KubernetesBuildTaskTest.java:59]: Not sure if these changes are related to this PR

**Comment by vikasrajputin** [comment]
I've rebased it, can you check now @rohanKanojia

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**